src/java.base/share/classes/sun/invoke/util/Wrapper.java

```diff
@@ -580,9 +580,9 @@ public Object wrap(int x) {
     }
 
     private static Number numberValue(Object x) {
-        if (x instanceof Number)     return (Number)x;
-        if (x instanceof Character)  return (int)(Character)x;
-        if (x instanceof Boolean)    return (Boolean)x ? 1 : 0;
+        if (x instanceof Number n)     return n;
```

Comment:
```java
return switch (x) {
    case Number n -> n;
    case Character c -> (int) c;
    case Boolean b -> b ? 1 : 0;
    // Remaining allowed case of void: Must be a null reference.
    case null, default -> (Number) x;
};
```

Reply:
This is used by Wrapper.convert and MethodHandles.insertArgumentPrimitive, which I don't think is a safe candidate either.